tests/FakeItEasy.Specs/UnconfiguredFakeSpecs.cs

```diff
@@ -166,10 +179,26 @@ public static void EqualsDifferentInstanceOfSameFakedType(FakedClass fake1, Fake
             "When I compare the instances using Equals"
                 .x(() => equalsResult = fake1.Equals(fake2));
 
-            "Then they compares as unequal"
+            "Then they compare as unequal"
                 .x(() => equalsResult.Should().BeFalse());
         }
 
+        [Scenario]
+        public static void GetHashCodeIsSameAsManager(FakedClass fake, int hashCode, int managerHashCode)
+        {
+            "Given a faked class instance"
+                .x(() => fake = A.Fake<FakedClass>());
+
+            "When I get its hash code"
+                .x(() => hashCode = fake.GetHashCode());
+
+            "And I get its manager's hash code"
+                .x(() => managerHashCode = Fake.GetFakeManager(fake).GetHashCode());
+
+            "Then the hash codes are the same"
+                .x(() => hashCode.Should().Be(managerHashCode));
+        }

```

Comment:
Isn't the fact that the fake and its manager have the same hashcode an implementation detail? i.e. do we really need a spec for it?

Reply:
You're right. I always struggle with how to test `GetHashCode`. We can make sure that fakes that are equal have the same hash code, but since we've a 1:1 map between fakes and their managers, the implemented behaviour devolves to "use reference semantics for `Equals` and `GetHashCode`". Then the only meaningful test I can think of is "a Fake generates the same hash code twice". Which isn't bad, but it is pretty weak.
I'd considered checking that two fakes give different hash codes, but that's not a requirement and the test could fail even when everything is working properly.

I can certainly add a test to ensure that the same fake's `GetHashCode` returns the same value more than once. And I can move the test back (and update it so it passes) if you like,  but the fact that we return the manager's hash code is meaningless enough that I'd almost rather remove the test altogether.

Have you a preference?

Reply:
Sorry, "remove the test that compares against the FakeManager's `GetHashCode` value".

Reply:
For now, "improved" the spec at least.

Reply:
> Have you a preference?

I think I'd remove this spec altogether.